//------------------------------------------------------------------------------
// protobuf specification for global objects. HAL's static global configuration
// comes from hal.json that HAL uses when it comes up. However, there are few
// knobs that are global to the NIC and can be changed after init time. Those
// are captured here. For example, NIC can boot in classic mode and then connect
// to Pensando controller that can make it trasition to flow mode.
//------------------------------------------------------------------------------

syntax = "proto3";

package device;
option go_package="halproto";

// NIC boots in classic mode first and may later transition to flow mode
enum DeviceMode {
  DEVICE_MODE_NONE             = 0;
  DEVICE_MODE_MANAGED_SWITCH   = 1;    // Managed NIC mode
  DEVICE_MODE_MANAGED_HOST_PIN = 2;    // Managed NIC mode 
  DEVICE_MODE_STANDALONE       = 3;    // flow based forwarding mode
}

// Global config object for NIC
message Device {
  DeviceMode    device_mode = 1;    // one of the operating modes
}
